src/Microsoft.PowerShell.Commands.Utility/commands/utility/new-object.cs

```diff
@@ -245,8 +245,15 @@ protected override void BeginProcessing()
                      "CannotFindAppropriateCtor",
                      ErrorCategory.ObjectNotFound, null));
             }
+#if !UNIX
             else // Parameterset -Com
             {
+                if (!Platform.IsWindowsDesktop)
+                {
+                    Exception exc = new NotSupportedException(NewObjectStrings.ComObjectPlatformIsNotSupported);
```

Comment:
COM works on NanoServer as well, as long as there it has a workable COM component
PS: the COM support was enabled when powershell core was still NanoServer powershell.

Reply:
So should we exclude the parameter only on Unix?

Reply:
The COM components we use in our tests are not available in NanoServer and IoT.
Yes, we should exclude the parameter only for Unix plats.

Reply:
Does "Shell.Application" work on NanoServer and IoT? can I use this in a test?

Reply:
Shell itself is not available on NanoServer and IoT. You can use it as a test, but just follow the existing tests in `test\powershell\engine\COM`.

Reply:
Why shouldn't we targeting our test to NanoServer and IoT?